File: spec/factories/sections.rb
FactoryBot.define do

  factory :section, class: Section do
    title  { Faker::Lorem.sentence(word_count: 3) }
    name  { Faker::Lorem.sentence(word_count: 2) }
    layout { Section::DEFAULT_SECTION_TITLE }
    show   { true }
    can_collapse_small { true }

    trait :with_items do
      after :create do |section|
        3.times do |count|
          embeddable = FactoryBot.create(:or_embeddable)
          section.page_items.create!(
            embeddable: embeddable,
            position: count + 1,
            column: PageItem::COLUMN_SECONDARY
          )
        end
      end
    end

    trait :on_page do
      association :interactive_page
    end
  end

end

File: lara-typescript/src/section-authoring/util/sections.ts
import { ISection } from "../api/api-types";

export const sectionName = ({position, title, name}: Pick<ISection, "position" | "title" | "name">) => {
  return name || `Section ${position} ${title ? title : ""}`;
};

export const sectionNumber = ({position, name}: Pick<ISection, "position" | "name">) => {
  return name || position;
};
